website/static/js/nodesPrivacySettingsTreebeard.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var ko = require('knockout');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
+
+
+function expandOnLoad() {
+    var tb = this;  // jshint ignore: line
+    for (var i = 0; i < tb.treeData.children.length; i++) {
+        var parent = tb.treeData.children[i];
+        tb.updateFolder(null, parent);
+        expandChildren(tb, parent.children);
+    }
+}
+
+function expandChildren(tb, children) {
+    var openParent = false;
+    for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        var parent = children[i].parent();
+        if (child.children.length > 0) {
+            expandChildren(tb, child.children);
+        }
+    }
+    if (openParent) {
+        openAncestors(tb, children[0]);
+    }
+}
+
+function openAncestors (tb, item) {
+    var parent = item.parent();
+    if(parent && parent.id > 0) {
+        tb.updateFolder(null, parent);
+        openAncestors(tb, parent);
+    }
+}
+
+function NodesPrivacyTreebeard(data, nodesState, nodesOriginal) {
+    /**
+     * nodesChanged and nodesState are knockout variables.  nodesChanged will keep track of the nodes that have
+     *  changed state.  nodeState is all the nodes in their current state.
+     * */
+    var tbOptions = $.extend({}, projectSettingsTreebeardBase.defaults, {
+        divID: 'grid',
```

Comment:
This is a very generic CSS ID; it's likely to be a duplicate with another ID on the page. Rather than hardcoding it here, better to pass it in as a argument


```js
new NodesPrivacyTreebeard('#nodesPrivacyTreebeard', ....);
```